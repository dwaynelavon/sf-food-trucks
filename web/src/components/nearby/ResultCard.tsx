import React from "react";

interface IProps {
    description: string;
    distance: string;
    location: string;
    title: string;
}

const ResultCard = ({ description, distance, location, title }: IProps) => (
    <div className="columns">
        <div className="column is-four-fifths">
            <div className="title is-size-4">{title}</div>
            <div className="subtitle is-size-5">{location}</div>
        </div>
        <div className="column">
            <div className="is-size-5">{distance} mi</div>
        </div>
    </div>
);

export default ResultCard;
