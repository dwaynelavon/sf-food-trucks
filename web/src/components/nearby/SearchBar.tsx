import React, { useCallback, useState } from "react";
import { SearchHandler } from "./types";

interface IProps {
    onSearch: SearchHandler;
}

const SearchBar = ({ onSearch }: IProps) => {
    const [lat, setLat] = useState<number>();
    const [lng, setLng] = useState<number>();

    const searchDisabled = !lat || !lng;

    // Change handlers
    const handleLatChange = useCallback(
        (e: React.ChangeEvent<HTMLInputElement>) => {
            setLat(parseFloat(e.target.value));
        },
        []
    );

    const handleLngChange = useCallback(
        (e: React.ChangeEvent<HTMLInputElement>) => {
            setLng(parseFloat(e.target.value));
        },
        []
    );

    // Search
    const handleSearch = useCallback(() => {
        // Search only happens when search is not disabled
        onSearch(lat!, lng!);
    }, [lat, lng]);

    const handleCurrentLocationSearch = useCallback(() => {
        navigator.geolocation.getCurrentPosition(
            (position: Position) => {
                onSearch(position.coords.latitude, position.coords.longitude);
            },
            () => {
                alert("can not find current location");
            }
        );
    }, []);

    return (
        <div className="field is-horizontal">
            <div className="field-body ">
                <div className="field">
                    <p className="control is-expanded has-icons-left">
                        <input
                            className="input"
                            type="number"
                            placeholder="Latitude"
                            onChange={handleLatChange}
                        />
                        <span className="icon is-small is-left">
                            <i className="fas fa-globe"></i>
                        </span>
                    </p>
                </div>
                <div className="field">
                    <p className="control is-expanded has-icons-left">
                        <input
                            className="input"
                            type="number"
                            placeholder="Longitude"
                            onChange={handleLngChange}
                        />
                        <span className="icon is-small is-left">
                            <i className="fas fa-globe"></i>
                        </span>
                    </p>
                </div>
                <div className="field is-grouped">
                    <p className="control">
                        <button
                            disabled={searchDisabled}
                            className="button is-info"
                            onClick={handleSearch}
                        >
                            Search
                        </button>
                    </p>
                    {navigator.geolocation && (
                        <p className="control">
                            <button
                                className="button is-dark"
                                onClick={handleCurrentLocationSearch}
                            >
                                Current Location
                            </button>
                        </p>
                    )}
                </div>
            </div>
        </div>
    );
};

export default SearchBar;
