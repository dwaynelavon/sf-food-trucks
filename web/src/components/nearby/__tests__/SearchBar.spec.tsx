import { shallow } from "enzyme";
import React from "react";
import ResultCard from "../ResultCard";
import SearchBar from "../SearchBar";

describe("Search Bar", () => {
    it("should render with default props", () => {
        const props = {
            onSearch: jest.fn(),
        };
        const wrapper = shallow(<SearchBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
