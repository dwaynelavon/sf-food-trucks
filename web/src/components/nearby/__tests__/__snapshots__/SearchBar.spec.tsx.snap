// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar should render with default props 1`] = `ShallowWrapper {}`;
